cpp/oneapi/dal/backend/interop/table_conversion.hpp

```diff
@@ -114,6 +115,105 @@ inline daal::data_management::NumericTablePtr convert_to_daal_table(const table&
     }
 }
 
+template <typename T>
+inline auto allocate_daal_csr_table(const T* data,
+                                    const std::size_t* column_indices,
+                                    const std::size_t* row_indices,
+                                    std::int64_t row_count,
+                                    std::int64_t column_count) {
+    return daal::data_management::CSRNumericTable::create(
+        data,
+        column_indices,
+        row_indices,
+        dal::detail::integral_cast<std::size_t>(column_count),
+        dal::detail::integral_cast<std::size_t>(row_count));
+}
+
+template <typename T>
+inline auto convert_to_daal_csr_table(array<T> data,
+                                      array<std::size_t> column_indices,
+                                      array<std::size_t> row_indices,
+                                      std::int64_t row_count,
+                                      std::int64_t column_count) {
+    if (!data.get_count() || !column_indices.get_count() || !row_indices.get_count())
+        return daal::services::SharedPtr<daal::data_management::CSRNumericTable>();
+    data.need_mutable_data();
```

Comment:
Will this step be copying?

Reply:
Depends on the `data`. If it contains mutable data - no.